### PR TITLE
[BPMSPL-744] GIT related enhancements for scalable workspaces

### PR DESCRIPTION
**Pull request by ederign**
JGITFS support for multiple and scalable FS
    (prerequisite for workspaces and cloud)
    Also:
    [AF-573] Refactoring in order to centralize FileSystem Calls
    [AF-569] Cleanup and simplify clustering

### CODE REVIEW

**Comment by ederign** [comment]
Depends on https://github.com/AppFormer/uberfire/pull/797

File: guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/git/GitMetadataStoreImpl.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
     private Logger logger = LoggerFactory.getLogger(GitMetadataStoreImpl.class);
     public static final String SEPARATOR = "/";
-    public static final String METADATA = "default://system/metadata";
```
Comment (ederign) [inline guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/git/GitMetadataStoreImpl.java]: @porcelli @manstis  I've to change metadata filesystem name due to namespace clash with system

**Comment by ederign** [comment]
jenkins retest this

**Comment by ederign** [comment]
jenkins retest this

File: guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/git/GitMetadataStoreImpl.java (hunk 1 above)
Comment (manstis) [reply guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/git/GitMetadataStoreImpl.java]: @ederign Is this "backwards" compatible with older installations? I don't mind if it recreates a new ```.niogit/metadata``` repository... as long as it doesn't break! @wmedvede IIRC you was asking about why ```metadata``` was under ```.niogit/system/metadata```? IDK if this change affects anything you was doing.. and if not you @wmedvede I wonder who it was!??!

**Pull request closed without merging**